insights/parsers/tests/test_ldif_config.py

```diff
@@ -0,0 +1,190 @@
+ s7directoryreplicasubtree || nsds7dirsynccookie || nsds7newwingroupsyncenable
+ d || nsds7newwinusersyncenabled || nsds7windowsdomain || nsds7windowsreplicas
+ ubtree || nsruvreplicalastmodified || nsstate || objectclass || onewaysync ||
+  winsyncdirectoryfilter || winsyncinterval || winsyncmoveaction || winsyncsub
+ treepair || winsyncwindowsfilter")(targetfilter = "(|(objectclass=nsds5Replic
+ a)(objectclass=nsds5replicationagreement)(objectclass=nsDSWindowsReplicationA
+ greement)(objectClass=nsMappingTree))")(version 3.0;acl "permission:Read Repl
+ ication Agreements";allow (compare,read,search) groupdn = "ldap:///cn=Read Re
+ plication Agreements,cn=permissions,cn=pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (targetattr=*)(version 3.0;acl "permission:Add Replication Agreements";al
+ low (add) groupdn = "ldap:///cn=Add Replication Agreements,cn=permissions,cn=
+ pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (targetattr=*)(targetfilter="(|(objectclass=nsds5Replica)(objectclass=nsd
+ s5replicationagreement)(objectclass=nsDSWindowsReplicationAgreement)(objectCl
+ ass=nsMappingTree))")(version 3.0; acl "permission:Modify Replication Agreeme
+ nts"; allow (read, write, search) groupdn = "ldap:///cn=Modify Replication Ag
+ reements,cn=permissions,cn=pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (targetattr=*)(targetfilter="(|(objectclass=nsds5replicationagreement)(ob
+ jectclass=nsDSWindowsReplicationAgreement))")(version 3.0;acl "permission:Rem
+ ove Replication Agreements";allow (delete) groupdn = "ldap:///cn=Remove Repli
+ cation Agreements,cn=permissions,cn=pbac,dc=idm,dc=nypd,dc=finest";)
+aci: (target = "ldap:///cn=meTo($dn),cn=*,cn=mapping tree,cn=config")(targetat
+ tr = "objectclass || cn")(version 3.0; acl "Allow hosts to read their replica
+ tion agreements"; allow(read, search, compare) userdn = "ldap:///fqdn=($dn),c
+ n=computers,cn=accounts,dc=idm,dc=nypd,dc=finest";)
+cn: mapping tree
+modifiersName: cn=Directory Manager
+modifyTimestamp: 20201026161600Z
+objectClass: top
+objectClass: extensibleObject
+numSubordinates: 3
+
+dn: cn=sasl,cn=config
+cn: sasl
+objectClass: top
+objectClass: nsContainer
+numSubordinates: 1
+"""
+
+LDIF_CONFIG_EMPTY = ""
+
+
+def test_ldip_parser():
+    ldif_config = LDIFParser(context_wrap(LDIF_CONFIG))
+    ldif_config = sorted(ldif_config, key=lambda x: x['dn'])
+    assert ldif_config[1]['dn'] == 'cn=changelog5,cn=config'
+    assert ldif_config[1]['createTimestamp'] == '20201026161228Z'
+    assert ldif_config[1]['creatorsName'] == 'cn=Directory Manager'
+    assert ldif_config[1]['nsslapd-changelogdir'] == '/var/lib/dirsrv/slapd-IDM-NYPD-FINEST/cldb'
+    assert ldif_config[1]['objectClass'] == 'extensibleobject'
+    assert ldif_config[2]['dn'] == 'cn=config'
+    assert ldif_config[2]['modifiersName'] == 'cn=directory manager'
+    assert ldif_config[2]['modifyTimestamp'] == '20210609192548Z'
+    assert ldif_config[2]['objectClass'] == 'nsslapdConfig'
+    assert ldif_config[6]['dn'] == 'cn=monitor'
```

Comment:
Here, I don't think the test is correct, you shouldn't break the original order of the 'dn' blocks in the file.

Reply:
I missed you have re-sorted the list above. Sorry about that.

But I don't think that re-sorted is required.

Reply:
Removed this check.